tests/refs/branches/lookup.c

```diff
@@ -23,17 +23,22 @@ void test_refs_branches_lookup__cleanup(void)
 void test_refs_branches_lookup__can_retrieve_a_local_branch(void)
 {
 	cl_git_pass(git_branch_lookup(&branch, repo, "br2", GIT_BRANCH_LOCAL));
+	cl_git_pass(git_branch_lookup(&branch, repo, "br2", GIT_BRANCH_ALL));
+	cl_git_fail(git_branch_lookup(&branch, repo, "br2", GIT_BRANCH_REMOTE));
 }
```

Comment:
You're leaking memory here, namely the results of the prior `git_branch_lookup` that are hidden with the new assignment to `branch`.  Probably best to just make each one of these calls their own test function

Reply:
I was scratching my head about that, thanks for the explanation :slightly_smiling_face:.